### ISSUE TITLE
Setting `notFound` does not result in 404

### ISSUE THREAD

**Issue by jillbucher-qz** [platypusrex/next-merge-props#7]
I noticed that when I'm setting `notFound`, the page returns a 200 status code.  `mergeProps` doesn't return`notFound` when one of the results sets it as true.  [This line](https://github.com/platypusrex/next-merge-props/blob/4e2a5c7b3b485e1b3e86dc132a2309633ddfbf48/src/utils/mergeResults.ts#L22) only checks for `redirect`, which means that it won't go into the block with `notFound` unless `redirect` is set on the result.

### Steps to reproduce
Go to a non-numeric id  (i.e. `/abcd`) and it will render TestPage with a 200 status code.
Expected: The page should go the 404 page and return a 404 response code.

```
import type { GetServerSidePropsContext } from 'next';
import { mergeProps } from 'next-merge-props';
import React from 'react';

export const getServerSideProps = mergeProps(
  [
    async (context: GetServerSidePropsContext) => {
      const { params } = context;
      const id = params?.id;
      if (typeof id === 'string' && id?.match(/^\d{4}$/)) {
        return { props: { id } };
      }
      return { props: {}, notFound: true };
    },
  ]
);

const TestPage = ({ id }) => {
  return (
    <div>
      <p>Test Page {id}</p>
    </div>
  );
};
export default TestPage;
```

**Comment by platypusrex** [comment]
Hey @jillbucher-qz! Great catch and thanks for pointing it out. Apologies for the late response as well. Super busy at the moment. It might take a few more days before I can get to this. I'd also be happy to accept a PR to address the fix if you feel up to it.

**Comment by platypusrex** [comment]
Hey again @jillbucher-qz! Apologies for the long wait here. This issue has been resolved and fix was published in v0.5.1. I'm going to close this out.
